### ISSUE TITLE
Access to XMLHttpRequest at 'https://mydomain.co.uk:9443/tests/_msearch?' from origin 'https://mydomain.co.uk' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.

### ISSUE THREAD

**Issue by manikumarrepuri** [appbaseio/reactivesearch#984]
**Issue Type:**
Issue

**Platform:**
Reactivesearch for web [ReactJS web application/ nodejs backend]

**Description:**
From a React frontend 
Code: <
ReactiveBase app="tests"
url= "https://mydomain.co.uk:9443"
>
Trying to connect but getting the below error. 

Access to XMLHttpRequest at 'https://mydomain.co.uk:9443/tests/_msearch?' from origin 'https://mydomain.co.uk' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.

This is my elasticsearch.yml on docker instance
-----------------------------------------------------
http.cors.enabled: true
http.cors.allow-credentials: true
http.cors.allow-origin: *
http.cors.allow-headers: X-Requested-With, X-Auth-Token, Content-Type, Content-Length, Authorization, Access-Control-Allow-Headers, Accept

**Comment by siddharthlatest** [comment]
@manikumarrepuri This looks fine. Do you have any other service (e.g. nginx, nodejs) sitting in front of your ElasticSearch? The same CORS rules need to be applied there as well.

**Comment by manikumarrepuri** [comment]
Thanks for the prmpt response @siddharthlatest , Yes I do have nodejs as backend and nginx as load balancer.
Find the CORS settings on the nodejs as below...

app.register(require('fastify-cors'), { 
    origin: true, //Access-Control-Allow-Origin
    methods: ['GET','PUT','POST','DELETE','PATCH','OPTIONS'], //Access-Control-Allow-Methods
    allowedHeaders: ['Origin', 'X-Requested-With', 'Content-Type', 'Accept', 'Access-Control-Allow-Origin', 
    'Cache-Control', 'Cookie', 'Accept', 'X-Requested-With'], //Access-Control-Allow-Headers
    credentials: true, //Access-Control-Allow-Credentials
    preflightContinue: true
});

Nginx as below - 443
---------------------------------------------------------------
server {
    listen 443;

    ssl on;
    server_name         {{my_fqdn}}, *.{{domain_suffix}};
    ssl_certificate     /etc/ssl/{{my_fqdn}}.crt;
    ssl_certificate_key /etc/ssl/{{my_fqdn}}.key;
    ssl_trusted_certificate /etc/ssl/my.cer;

    location / {
        include /etc/nginx/conf.d/api_proxy.conf;
        proxy_pass http://frontend$request_uri;
        proxy_redirect http://frontend $scheme://$host;
    }

    error_page 400 401 402 403 404 /404.html;
    location = /404.html {
    root /usr/share/nginx/html;
    internal;
    }

    error_page 500 501 502 503 504 /503.html;
    location = /503.html {
    root /usr/share/nginx/html;
    internal;
    }

    location = /error.css {
    root /usr/share/nginx/html;
    }

}
backend
---------------------------
server {
    listen {{backend_port}};

    ssl on;
    server_name         {{my_fqdn}}, *.{{domain_suffix}};
    ssl_certificate     /etc/ssl/{{my_fqdn}}.crt;
    ssl_certificate_key /etc/ssl/{{my_fqdn}}.key;
    ssl_trusted_certificate /etc/ssl/my.cer;

    location /api/ {
        include /etc/nginx/conf.d/api_proxy.conf;
        proxy_pass http://backend$request_uri;
        proxy_redirect http://backend $scheme://$host;
    }
}

**Comment by manikumarrepuri** [comment]
> @manikumarrepuri This looks fine. Do you have any other service (e.g. nginx, nodejs) sitting in front of your ElasticSearch? The same CORS rules need to be applied there as well.

**Comment by manikumarrepuri** [comment]
By mistake, clicked close button...Appreciate your response.

**Comment by vincepich** [comment]
Put this in your configuration file _elastichsearch.yml_:

http.cors.enabled: true
http.cors.allow-credentials: true
http.cors.allow-origin: "*"
**http.cors.allow-methods: OPTIONS, HEAD, GET, POST, PUT, DELETE**
http.cors.allow-headers: X-Requested-With, X-Auth-Token, Content-Type, Content-Length, Authorization, Access-Control-Allow-Headers, Accept

**Comment by manikumarrepuri** [comment]
@siddharthlatest 
I have enabled cors settings on nginx side, looks like the issue is vanished but facing another issue.

react-dom.production.min.js:4408 Invariant Violation: Minified React error #31; visit https://reactjs.org/docs/error-decoder.html?invariant=31&args[]=object%20with%20keys%20%7Bvalue%2C%20relation%7D&args[]= for the full message or use the non-minified dev environment for full errors and additional helpful warnings.

ReactiveBase.js:130 An error has occured. You're using Reactivesearch Version: 3.0.0-rc.9. If you think this is a problem with Reactivesearch, please try updating to the latest version. If you're already at the latest version, please open an issue at https://github.com/appbaseio/reactivesearch/issues

**Comment by siddharthlatest** [comment]
@manikumarrepuri Since this is a separate issue, can you open it in a new issue thread? Also, if you just started seeing this right now, then you can also try downgrading the library version, that would be helpful to know if it's specific to this particular version.

**Comment by manikumarrepuri** [comment]
Makes sense. closing this issue.
Below is the code added to nginx location /
add_header 'Access-Control-Allow-Origin' '*';
add_header 'Access-Control-Allow-Methods' 'GET,PUT,POST,DELETE,PATCH,OPTIONS';
add_header 'Access-Control-Allow-Headers' 'Origin, X-Requested-With, Content-Type, Accept, Access-Control-Allow-Origin, Cache-Control, Cookie, Accept, X-Requested-With';
